tokens/figma/update-skins.mjs

```diff
@@ -0,0 +1,310 @@
+          scopes: variableScopes,
+        });
+
+        newData.variableModeValues.push({
+          action: "UPDATE",
+          variableId: existingVariable.id,
+          modeId: existingMode,
+          value: variableValue,
+        });
+      } else {
+        const tempId = generateTempId(
+          variableName,
+          collectionName
+        );
+
+        const collectionId = Object.values(
+          existingCollections
+        ).find(
+          (collection) =>
+            collection.name === collectionName
+        ).id;
+
+        newData.variables.push({
+          action: "CREATE",
+          id: tempId,
+          name: variableName,
+          variableCollectionId: collectionId,
+          resolvedType: variableType,
+          scopes: variableScopes,
+        });
+
+        newData.variableModeValues.push({
+          action: "CREATE",
+          variableId: tempId,
+          modeId: existingMode,
+          value: variableValue,
+        });
+
+        allVariableNamesInCurrentData.add(
+          variableName
+        );
+      }
+    }
+
+    const variableGroups = [
+      {
+        variables: jsonData[brand].palette,
+        collectionName: "Palette",
+        resolvedType: "COLOR",
+        variableScopes: ["ALL_SCOPES"],
+      },
+    ];
+
+    variableGroups.forEach(
+      ({
+        variables,
+        collectionName,
+        resolvedType,
+        variableScopes,
+      }) => {
+        variables.forEach((variable) => {
+          updateVariables(
+            variable.name,
+            variable.value,
+            collectionName,
+            existingVariables,
+            existingCollections,
+            resolvedType,
+            variableScopes,
+            allVariableNamesInCurrentData
+          );
+        });
+      }
+    );
+
+    return newData;
+  } catch (error) {
+    console.error("Error:", error);
+    throw error; // rethrow the error to be handled later
+  }
+}
+
+async function postCollections(
+  brand,
+  FILE_KEY,
+  FIGMA_TOKEN
+) {
+  try {
+    const newData = await updateCollections(
+      collectionNames,
+      FILE_KEY,
+      FIGMA_TOKEN
+    );
+
+    const response = await fetch(
+      `https://api.figma.com/v1/files/${FILE_KEY}/variables/`,
+      {
+        method: "POST",
+        headers: {
+          "X-Figma-Token": FIGMA_TOKEN,
+          "Content-Type": "application/json",
+        },
+        body: JSON.stringify(newData),
+      }
+    );
+
+    const data = await response.json();
+    console.log(
+      `Success creating collections for brand ${brand}:`,
+      data
+    );
+  } catch (error) {
+    console.error(
+      `Error creating collections for for brand ${brand}:`,
+      error
+    );
+  }
+}
+
+async function postPalette(
+  jsonData,
+  brand,
+  FILE_KEY,
+  FIGMA_TOKEN
+) {
+  try {
+    const newData = await updatePalette(
+      jsonData,
+      brand,
+      FILE_KEY,
+      FIGMA_TOKEN
+    );
+
+    const response = await fetch(
```

Comment:
this piece of code is repeated a lot. There is not a way to encapsulate it in a function or similar? 🧐

In this file and others

Reply:
Agree with @yceballost, this can be modularized. Maybe a /utils folder where we can put all the auxiliary functions so that we can reuse them.

Reply:
I've made a big refactor to avoid repeating this codeand also improvements in file organization